### PR TITLE
[CARBONDATA-770] Fixed Not null filter query data mismatch issue

### PR DESCRIPTION
**Pull request by kumarvishal09**
Problem: Not null filter query is selecting null values.
Solution: Problem is while parsing the data based on data type we are not parsing for int, double, float, and long data type, need to add case for the same

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1128/

**Comment by ravipesala** [comment]
@kumarvishal09 Please add the testcase.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1135/

**Comment by kumarvishal09** [comment]
added testcase.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1137/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1138/

File: core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java [hunk 1]
```diff
@@ -507,12 +507,25 @@ public static String parseValue(String value, CarbonDimension dimension) {
...
+          Long.parseLong(value);
+          break;
+        case FLOAT:
+          Float.parseFloat(value);
+          break;
         default:
-          return value;
+          // do nothing
       }
     } catch (Exception e) {
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java]: better to change to `NumberFormatException`

File: core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java [hunk 2]
```diff
@@ -507,12 +507,25 @@ public static String parseValue(String value, CarbonDimension dimension) {
...
       return null;
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java]: if null is returned, is it recorded in bad record log?

Comment (kumarvishal09) [inline core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java]: yes

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1152/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**